zerver/views/home.py

```diff
@@ -118,8 +121,16 @@ def home(request: HttpRequest) -> HttpResponse:
 
     return hello_view(request)
 
-@zulip_login_required
 def home_real(request: HttpRequest) -> HttpResponse:
+    # Add client info.
+    # This was previously done as a part of @zulip_login_required.
+    # Since, we allow anonymous users now, we do it manually here.
+    # TODO: rate limit all users.
+    client_name = get_client_name(request)
+    if not client_name.startswith("Zulip"):
+        client_name = "website"
+    request.client = get_client(client_name)
```

Comment:
Would `@add_logging_data` work instead of this logic?  Would require patching this:

```
    request.client = get_client(client_name)                                                         
-    if not skip_update_user_activity:                                                                
-    if not skip_update_user_activity and user_profile.is_authenticated:
        update_user_activity(request, user_profile, query)                                           
```

(Since we'll get the AnonymousUser here, and there's no analogue of that step we want to do for the anonymous case).